docs/sources/flow/reference/components/loki.source.api.md

```diff
@@ -0,0 +1,93 @@
+
+## Usage
+
+```river
+loki.source.api LABEL {
+    http {
+        listen_address = "LISTEN_ADDRESS"
+        listen_port = PORT 
+    }
+    forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.api` supports the following arguments:
+
+ Name                     | Type                 | Description                                                | Default | Required 
+--------------------------|----------------------|------------------------------------------------------------|---------|----------
+ `forward_to`             | `list(LogsReceiver)` | List of receivers to send log entries to.                  |         | yes      
+ `use_incoming_timestamp` | `bool`               | Whether or not to use the timestamp received from request. | `false` | no       
+ `labels`                 | `map(string)`        | The labels to associate with each received logs record.    | `{}`    | no       
+ `relabel_rules`          | `RelabelRules`       | Relabeling rules to apply on log entries.                  | `{}`    | no       
+
+The `relabel_rules` field can make use of the `rules` export value from a
+[`loki.relabel`][loki.relabel] component to apply one or more relabeling rules to log entries before they're forwarded to the list of receivers in `forward_to`.
+
+[loki.relabel]: {{< relref "./loki.relabel.md" >}}
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.source.heroku`:
+
+ Hierarchy | Name     | Description                                        | Required 
+-----------|----------|----------------------------------------------------|----------
+ `http`    | [http][] | Configures the HTTP server that receives requests. | no       
+
+[http]: #http
+
+### http
+
+{{< docs/shared lookup="flow/reference/components/loki-server-http.md" source="agent" >}}
+
+## Exported fields
+
+`loki.source.api` does not export any fields.
+
+## Component health
+
+`loki.source.api` is only reported as unhealthy if given an invalid configuration.
+
+## Debug metrics
+
+The following are some of the metrics that are exposed when this component is used. Note that the metrics include labels such as `status_code` where relevant, which can be used to measure request success rates.
+
+* `agent_loki_source_api_request_duration_seconds` (histogram): Time (in seconds) spent serving HTTP requests.
+* `agent_loki_source_api_request_message_bytes` (histogram): Size (in bytes) of messages received in the request.
+* `agent_loki_source_api_response_message_bytes` (histogram): Size (in bytes) of messages sent in response.
+* `agent_loki_source_api_tcp_connections` (gauge): Current number of accepted TCP connections.
+
+## Example
+
+This example starts an HTTP server on `0.0.0.0` address and port `9999`. The server receives log entries and forwards them to a `loki.echo` component while adding a `forwarded="true"` label.
+
+```river
+loki.echo "print" {}
```

Comment:
nit: could we forward to a loki.write instead and amend the previous sentence? 

I think it will be more useful for people who are looking to copy/paste from our docs

```
loki.write "local" {
  endpoint {
    url = "loki:3100/api/v1/push"
    basic_auth {
      username = env("LOKI_USERNAME")
      password = env("LOKI_PASSWORD")
    }
  }
}
```